shell/components/NavigationSecondary.vue

```diff
@@ -0,0 +1,555 @@
+    },
+
+    supportLink() {
+      const product = this.$store.getters['currentProduct'];
+
+      if (product?.supportRoute) {
+        return { ...product.supportRoute, params: { ...product.supportRoute.params, cluster: this.clusterId } };
+      }
+
+      return { name: `c-cluster-${ product?.name }-support` };
+    },
+
+    displayVersion() {
+      if (this.isSingleProduct?.getVersionInfo) {
+        return this.isSingleProduct?.getVersionInfo(this.$store);
+      }
+
+      const { displayVersion } = getVersionInfo(this.$store);
+
+      return displayVersion;
+    },
+
+    allNavLinks() {
+      if ( !this.clusterId || !this.$store.getters['cluster/schemaFor'](UI.NAV_LINK) ) {
+        return [];
+      }
+
+      return this.$store.getters['cluster/all'](UI.NAV_LINK);
+    },
+
+    counts() {
+      const managementReady = this.managementReady;
+      const product = this.$store.getters['currentProduct'];
+
+      if ( !managementReady || !product ) {
+        return {};
+      }
+
+      const inStore = product.inStore;
+
+      // So that there's something to watch for updates
+      if ( this.$store.getters[`${ inStore }/haveAll`](COUNT) ) {
+        const counts = this.$store.getters[`${ inStore }/all`](COUNT)[0].counts;
+
+        return counts;
+      }
+
+      return {};
+    },
+
+    allSchemas() {
+      const managementReady = this.managementReady;
+      const product = this.$store.getters['currentProduct'];
+
+      if ( !managementReady || !product ) {
+        return [];
+      }
+
+      return this.$store.getters[`${ product.inStore }/all`](SCHEMA);
+    },
+
+    namespaces() {
+      return this.$store.getters['activeNamespaceCache'];
+    },
+
+    favoriteTypes: mapPref(FAVORITE_TYPES),
+  },
+  watch: {
+    counts(a, b) {
+      if ( a !== b ) {
+        this.queueUpdate();
+      }
+    },
+
+    allSchemas(a, b) {
+      if ( a !== b ) {
+        this.queueUpdate();
+      }
+    },
+
+    allNavLinks(a, b) {
+      if ( a !== b ) {
+        this.queueUpdate();
+      }
+    },
+
+    favoriteTypes(a, b) {
+      if ( !isEqual(a, b) ) {
+        this.queueUpdate();
+      }
+    },
+
+    locale(a, b) {
+      if ( !isEqual(a, b) ) {
+        this.getGroups();
+      }
+    },
+
+    namespaceMode(a, b) {
+      if ( !isEqual(a, b) ) {
+        // Immediately update because you'll see it come in later
+        this.getGroups();
+      }
+    },
+
+    namespaces(a, b) {
+      if ( !isEqual(a, b) ) {
+        // Immediately update because you'll see it come in later
+        this.getGroups();
+      }
+    },
+
+    clusterReady(a, b) {
+      if ( !isEqual(a, b) ) {
+        // Immediately update because you'll see it come in later
+        this.getGroups();
+      }
+    },
+
+    product(a, b) {
+      if ( !isEqual(a, b) ) {
+        // Immediately update because you'll see it come in later
+        this.getGroups();
+      }
+    },
```

Comment:
What does the `product` watcher operate on? I don't see a corresponding `product` in `data` or `computed`.